solvers/MotionPlanningTree.m

```diff
@@ -0,0 +1,273 @@
+        if options.visualize && (mod(obj.n,options.display_after_every)==0 || (try_goal && status == obj.REACHED))
+          obj = obj.drawTree(last_drawn_edge_num);
+          last_drawn_edge_num = obj.n-1;
+        end
+
+        if try_goal && status == obj.REACHED
+          info = 1;
+          path_ids = obj.getPathToVertex(obj.n);
+          if options.visualize
+            drawPath(obj, path_ids);
+          end
+          break;
+        end
+      end
+      path_ids = obj.getPathToVertex(obj.n);
+    end
+
+    function [TA, TB,  path_ids_A, path_ids_B, info] = rrtConnect(TA, x_start, x_goal, TB, options)
+      if nargin < 5, options = struct(); end
+      defaultOptions.display_after_every = 50;
+      defaultOptions.max_length_between_constraint_checks = 0.001;
+      defaultOptions.goal_bias = .05;
+      defaultOptions.N = 10000;
+      defaultOptions.visualize = false;
+      options = applyDefaults(options, defaultOptions);
+      assert(TA.isValidConfiguration(x_start))
+      TA = TA.init(x_start);
+      if nargin < 4 || isempty(TB) 
+        TB = TA;
+        assert(TB.isValidConfiguration(x_goal))
+        TB = TB.init(x_goal);
+      else
+        typecheck(TB, 'MotionPlanningTree');
+        assert(TB.isValidConfiguration(x_goal))
+        TB = TB.init(x_goal);
+      end
+      info = 2;
+      last_drawn_edge_num_A = 1;
+      last_drawn_edge_num_B = 1;
+      for i = 1:options.N
+        if mod(i,2) == 0
+          [TA, TB, path_ids_A, path_ids_B] = rrtConnectIteration(TA, TB, options.goal_bias);
+        else
+          [TB, TA, path_ids_B, path_ids_A] = rrtConnectIteration(TB, TA, options.goal_bias);
+        end
+        if options.visualize && (TA.n + TB.n - last_drawn_edge_num_A - last_drawn_edge_num_B > options.display_after_every || ~isempty(path_ids_A))
+          TA = TA.drawTree(last_drawn_edge_num_A);
+          TB = TB.drawTree(last_drawn_edge_num_B);
+          drawnow
+          last_drawn_edge_num_A = TA.n-1;
+          last_drawn_edge_num_B = TB.n-1;
+        end
+        if ~isempty(path_ids_A)
+          assert(~isempty(path_ids_B))
+          info = 1;
+          if options.visualize
+            drawPath(TA, path_ids_A, TB, path_ids_B);
+          end
+          break;
+        end
+      end
+    end
+
+    function [TA, TB, path_ids_A, path_ids_B] = rrtConnectIteration(TA, TB, goal_bias)
+      x_sample = TA.randomConfig();
+      [TA, status, id_new] = extend(TA, x_sample);
+%       [TA, status, id_new] = connect(TA, x_sample);
+      path_ids_A = [];
+      path_ids_B = [];
+      if status == TA.TRAPPED && ~isempty(id_new), keyboard; end
+      if status ~= TA.TRAPPED && rand < goal_bias
+        [TB, status, id_last] = connect(TB, TA.getVertex(id_new));
+
+        if status == TB.REACHED
+          path_ids_A = TA.getPathToVertex(id_new);
+          path_ids_B = TB.getPathToVertex(id_last);
+        end
+      end
+    end
+
+
+    function [obj, q_new, id_near] = newConfig(obj, q)
```

Comment:
i'm ok with that.
On Feb 16, 2015, at 2:46 PM, Andres Valenzuela <notifications@github.com> wrote:

> In solvers/MotionPlanningTree.m:
> 
> > +%       [TA, status, id_new] = connect(TA, x_sample);
> > +      path_ids_A = [];
> > +      path_ids_B = [];
> > +      if status == TA.TRAPPED && ~isempty(id_new), keyboard; end
> > +      if status ~= TA.TRAPPED && rand < goal_bias
> > +        [TB, status, id_last] = connect(TB, TA.getVertex(id_new));
> > +
> > +        if status == TB.REACHED
> > +          path_ids_A = TA.getPathToVertex(id_new);
> > +          path_ids_B = TB.getPathToVertex(id_last);
> > +        end
> > +      end
> > +    end
> > +
> > +
> > +    function [obj, q_new, id_near] = newConfig(obj, q)
> newPoint?
> 
> —
> Reply to this email directly or view it on GitHub.
>